tensornetwork/network_components.py

```diff
@@ -46,36 +46,36 @@ class Node:
   an arbitrary dimension.
   """
 
-  def __init__(self, tensor: Tensor, name: Text, axis_names: List[Text],
-               network: TensorNetwork) -> None:
-    """Create a node for the TensorNetwork.
+  def __init__(self, name: Text, axis_names: List[Text], network: TensorNetwork,
+               shape: Optional[Tuple[int]] = None) -> None:
```

Comment:
I'm not sure `Optional` makes sense here. Is there ever a case where the shape of a node should be unknown?

Reply:
Actually nevermind. I now see that `node.shape` is a property.

Reply:
Yeah, I was doubting myself for a while, but with raising an error, I think this is ok. Also note this is an abstract class and cannot be directly instantiated anyhow.